ensembleperturbation/perturbation/make_storm_ensemble.py

```diff
@@ -0,0 +1,765 @@
+        # convert from nautical miles to statute miles
+        rmw_sm = rmax * UNIT_CONVERSIONS['nautical_mile']['statute_mile']
+        if rmw_sm < 15:
+            return '<15sm'  # very small
+        elif rmw_sm < 25:
+            return '15-25sm'  # small
+        elif rmw_sm < 35:
+            return '25-35sm'  # medium
+        elif rmw_sm < 45:
+            return '35-45sm'  # large
+        else:
+            return '>45sm'  # very large
+
+
+def utm_proj_from_lon(lon_mean: float) -> Proj:
+    """
+    utm_from_lon - UTM zone for a longitude
+    Not right for some polar regions (Norway, Svalbard, Antartica)
+    :param lon_mean: longitude
+
+    :usage   x_utm,y_utm   = myProj(lon, lat  , inverse=False)
+    :usage   lon, lat      = myProj(xutm, yutm, inverse=True)
+    """
+
+    zone = floor((lon_mean + 180) / 6) + 1
+    # print("Zone is " + str(zone))
+
+    return Proj(f'+proj=utm +zone={zone}K, +ellps=WGS84 +datum=WGS84 +units=m +no_defs')
+
+
+def get_offset(x1: float, y1: float, x2: float, y2: float, d: float,) -> (float, float):
+    """
+    get_offset(x1,y1,x2,y2,d)
+      - get the perpendicular offset to the line (x1,y1) -> (x2,y2) by a distance of d
+    """
+    if x1 == x2:
+        dx = d
+        dy = 0
+    elif y1 == y2:
+        dy = d
+        dx = 0
+    else:
+        # if z is the distance to your parallel curve,
+        # then your delta-x and delta-y calculations are:
+        #   z**2 = x**2 + y**2
+        #   y = pslope * x
+        #   z**2 = x**2 + (pslope * x)**2
+        #   z**2 = x**2 + pslope**2 * x**2
+        #   z**2 = (1 + pslope**2) * x**2
+        #   z**2 / (1 + pslope**2) = x**2
+        #   z / (1 + pslope**2)**0.5 = x
+
+        # tangential slope approximation
+        slope = (y2 - y1) / (x2 - x1)
+
+        # normal slope
+        pslope = -1 / slope  # (might be 1/slope depending on direction of travel)
+        psign = ((pslope > 0) == (x1 > x2)) * 2 - 1
+
+        dx = psign * d / sqrt(1 + pslope ** 2)
+        dy = pslope * dx
+
+    return dx, dy
+
+
+if __name__ == '__main__':
+    ##################################
+    # Example calls from command line for 2018 Hurricane Florence:
+    # - python3 make_storm_ensemble.py 3 al062018 2018-09-11-06 2018-09-17-06
+    # - python3 make_storm_ensemble.py 5 Florence2018 2018-09-11-06
+    ##################################
+
+    # Implement argument parsing
+    argument_parser = ArgumentParser()
+    argument_parser.add_argument('number_of_perturbations', help='number of perturbations')
+    argument_parser.add_argument('storm_code', help='storm name/code')
+    argument_parser.add_argument('start_date', nargs='?', help='start date')
+    argument_parser.add_argument('end_date', nargs='?', help='end date')
+    arguments = argument_parser.parse_args()
+
+    # hardcoding variable list for now
+    variables = [
+        MaximumSustainedWindSpeed,
+        RadiusOfMaximumWinds,
+        AlongTrack,
+        CrossTrack,
+    ]
```

Comment:
use class-based variable system to store all the per-variable data (for instance, `AlongTrack` contains the column name, lower and upper bounds, perturbation randomness type, and historical errors)